PR for issue #139 and #140 (and some more) (#144)
eclipse/microprofile-graphql: spec/src/main/asciidoc/entities.asciidoc

getter method, setter method or field. The placement will determine whether it applies to the type, input type or both,
respectively.

Example, placing the annotation on the field: 

[source,java,numbered]
----
public class Item {
    // ...
    @NonNull
    private String name;
    // ....
----

Will result in both the Type and Input to be marked not null (!):

[source,graphql,numbered]
----
input ItemInput {
  name: String!
  # ...
}

type Item {
  name: String!
  #...  
}
----

The annotation can also be use to indicate that elements in a collection can not be null, example:

[source,java,numbered]
----
public class SuperHero {
    // ...
    private List<@NonNull String> superPowers;
    // ...
}
----

> This indicated that superPowers can be null, but if it's not, it has to have elements in the list.

Reviewer: ```suggestion
This indicated that superPowers can be null, but if it's not, then it must only contain non-null entries.
```

This might require some clarification, but my understanding of `List<@NonNull String> superPowers` was that it could not have any null entries in it - which is different than non-empty.

So assuming `List<@NonNull String> superPowers`, which of the following are valid?
1) ["Laser eyes", "Super breath"] - valid - non-empty, and both values are non-null
2) ["Spidey sense", null] - contains one non-null entry, and one null entry - I think this is not valid
3) [] - empty, but no nulls - I'm not sure about this case, but I think it might be valid
4) [null] - one entry, null - I think this is not valid.
5) [""] - non-empty list with non-null (though empty) entry - I think this is valid

Author: Agree. Done

> Placing a `@NonNull` on the List can also makes the actual list non null;

Reviewer: I think this line should go after the example to avoid confusion (when I first read this, I was expecting the following example to show that both the list itself could not be null and the list entries could not be null).

Author: Done
